From 199227ae504a01e71a0e66b3fe46f47899ca762b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Thu, 21 Apr 2016 02:15:54 +0200
Subject: [PATCH] Allow specify gpg key for a repository
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Quite hacky way, but current command line syntax doesn't support
additional per-repository settings.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
Signed-off-by: Frédéric Pierret (fepitre) <frederic.epitre@orange.fr>
---
 src/pylorax/cmdline.py |  4 ++--
 src/pylorax/dnfbase.py | 10 ++++++++++
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/src/pylorax/cmdline.py b/src/pylorax/cmdline.py
index 652ffbbd..e755ee57 100644
--- a/src/pylorax/cmdline.py
+++ b/src/pylorax/cmdline.py
@@ -36,7 +36,7 @@ def lorax_parser(dracut_default=""):
     required.add_argument("-p", "--product", help="product name", required=True, metavar="PRODUCT")
     required.add_argument("-v", "--version", help="version identifier", required=True, metavar="VERSION")
     required.add_argument("-r", "--release", help="release information", required=True, metavar="RELEASE")
-    required.add_argument("-s", "--source", help="source repository (may be listed multiple times)",
+    required.add_argument("-s", "--source", help="source repository (may be listed multiple times), append gpgkey URL in brackets to enable package verification",
                         metavar="REPOSITORY", action="append", default=[])
     required.add_argument("--repo", help="source dnf repository file", type=os.path.abspath,
                           dest="repos", metavar="REPOSITORY", action="append", default=[])
@@ -44,7 +44,7 @@ def lorax_parser(dracut_default=""):
     # optional arguments
     optional = parser.add_argument_group("optional arguments")
     optional.add_argument("-m", "--mirrorlist",
-                        help="mirrorlist repository (may be listed multiple times)",
+                        help="mirrorlist repository (may be listed multiple times), append gpgkey URL in brackets to enable package verification",
                         metavar="REPOSITORY", action="append", default=[])
     optional.add_argument("-t", "--variant", default="",
                         help="variant name", metavar="VARIANT")
diff --git a/src/pylorax/dnfbase.py b/src/pylorax/dnfbase.py
index c4d514f1..3404ddd0 100644
--- a/src/pylorax/dnfbase.py
+++ b/src/pylorax/dnfbase.py
@@ -130,6 +130,11 @@ def get_dnf_base_object(installroot, sources, mirrorlists=None, repos=None,
             continue
         repo_name = "lorax-repo-%d" % i
         repo = dnf.repo.Repo(repo_name, conf)
+        if '(' in r and ')' in r:
+            assert r[-1] == ')'
+            r, gpgkey = r[:-1].split('(')
+            repo.gpgkey = [gpgkey]
+            repo.gpgcheck = True
         repo.baseurl = [r]
         if proxy:
             repo.proxy = proxy
@@ -150,6 +155,11 @@ def get_dnf_base_object(installroot, sources, mirrorlists=None, repos=None,
             continue
         repo_name = "lorax-mirrorlist-%d" % i
         repo = dnf.repo.Repo(repo_name, conf)
+        if '(' in r and ')' in r:
+            assert r[-1] == ')'
+            r, gpgkey = r[:-1].split('(')
+            repo.gpgkey = [gpgkey]
+            repo.gpgcheck = True
         repo.mirrorlist = r
         if proxy:
             repo.proxy = proxy
-- 
2.25.1

